public/app/features/plugins/admin/api.ts

```diff
@@ -20,14 +20,16 @@ export async function getCatalogPlugin(id: string): Promise<CatalogPlugin> {
 }
 
 export async function getPluginDetails(id: string): Promise<CatalogPluginDetails> {
-  const localPlugins = await getLocalPlugins();
-  const local = localPlugins.find((p) => p.id === id);
-  const isInstalled = Boolean(local);
-  const [remote, versions, localReadme] = await Promise.all([
-    getRemotePlugin(id, isInstalled),
-    getPluginVersions(id),
+  const remote = await getRemotePlugin(id);
+  const isPublished = Boolean(remote);
+
+  const [localPlugins, versions, localReadme] = await Promise.all([
+    getLocalPlugins(),
+    getPluginVersions(id, isPublished),
     getLocalPluginReadme(id),
   ]);
+
+  const local = localPlugins.find((p) => p.id === id);
```

Comment:
**Suggestion:** maybe we could extract the logic of fetching a local plugin into a separate function

(as far as I can see it is also happening in the `getPlugin()`)

E.g.:
```javascript
async function getLocalPlugin(id: string): Promise<LocalPlugin[]> {
    const localPlugins = await getLocalPlugins();
    
    return localPlugins.find((p) => p.id === id);
}
```